src/words/environment/Environment.java

```diff
@@ -127,19 +127,32 @@ public int getScopeDepth() {
 	 * @throws WordsRuntimeException
 	 */
 	public WordsObject createObject(String objectName, String className, Position position) throws WordsRuntimeException {
-		Property property = getVariable(objectName);
+		WordsClass wordsClass = getClass(className);
+		WordsObject newObject = new WordsObject(objectName, wordsClass, position);
+		createLocalVariable(objectName, new Variable(newObject));
 		
-		if (property.type == Property.PropertyType.NOTHING) {
-			WordsClass wordsClass = getClass(className);
-			
-			WordsObject newObject = new WordsObject(objectName, wordsClass, position);
-			getCurrentScope().variables.put(objectName, new Property(newObject));
-			
-			objectsByClass.get(wordsClass).add(newObject);
-			
-			return newObject;
+		return newObject;
+	}
+	
+	/**
+	 * Creates a new local variable in the environment. If a variable of the same name already exists, 
+	 * it will overwrite the existing variable unless the existing variable is an object.
+	 * When successful, the variable is added to the current scope
+	 * @throws ObjectAlreadyExistsException
+	 */
+	public void createLocalVariable(String varName, Variable value) throws ObjectAlreadyExistsException {
+		Variable existingVariable = getVariable(varName);
+		if (existingVariable.type == Variable.VariableType.OBJECT) {
+			throw new ObjectAlreadyExistsException(varName);
+		} else if (existingVariable.type != Variable.VariableType.NOTHING) {
+			existingVariable.copyOtherVariable(value);
 		} else {
-			throw new ObjectAlreadyExistsException(objectName);
+			getCurrentScope().variables.put(varName, value);
+		}
+		
+		// If this variable is an object, also add it to objects by class
+		if (value.type == Variable.VariableType.OBJECT) {
```

Comment:
This will throw a null pointer exception if value is null, and it would be null if INodeCreateLocalVariable is parsing a statement like "x is nothing" or "x is Fred".